README.md

```diff
@@ -18,6 +10,38 @@ Current assumptions / changes to the rules made to simplify the gameplay:
 - Variable number of players (as few as 2 players)
 - Subsets of the full deck can be used (as few as 2 cards per player)
 
+### Reinforcement Learning
+
+I'm using SARSA as my reinforcement learning front end. This means that:
+ - The game is played until it's the agent's turn
+ - The agent then has its current state
+   - This is *current hand* and *current trick*
+ - The network learns
+   - See *Updating the Q-values and Neural Network*
+ - The agent chooses an action
+   - Currently a mostly-greedy choice based on Q values
+ - The rest of the trick is played out
+ - The reward is calculated
+
+### Updating the Q-values and Neural Network
+
+Once a new state has been found:
+ - Work out *best* possible Q value for current state/action pair
+ - Uses this value, plus any rewards, to recalculate the Q value for the previous state/action pair
+ - Use this new value of the previous state/action pair to train the neural network once
+
+### Neural Network
+
+This is currently a very basic network with:
+ - 156 input nodes
+   - 52 to represent the cards in the agent's hand
+   - 52 to represent the cards already in the trick
+   - 52 to represent the card being played
```

Comment:
Could you thin this down to 52 total input nodes, with each node having specific values for the locations each card can be:

- in hand
- in trick
- being played
- elsewhere